pandas/_libs/sparse.pyx

```diff
@@ -86,17 +83,15 @@ cdef class IntIndex(SparseIndex):
         if self.npoints == 0:
             return
 
-        if min(self.indices) < 0:
+        if self.indices.min() < 0:
             raise ValueError("No index can be less than zero")
 
-        if max(self.indices) >= self.length:
+        if self.indices.max() >= self.length:
             raise ValueError("All indices must be less than the length")
 
-        for index in self.indices:
```

Comment:
The buffer access may need to be optimized, but in principle this should be faster in many cases because it can short-circuit.

Reply:
It will only short-circuit in the case of an integrity failure, which should be the minority of cases.